File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Controllers/TableViewControllers/TableViewCells/BtTextViewTableViewCell.cs
using System;

using Foundation;
using UIKit;

namespace bugTrapKit
{
	public partial class BtTextViewTableViewCell : BtNewBugDetailsTableViewCell, IUITextViewDelegate
	{

		public UITextView TxtView {
			get { return TextView; }
		}

		public BtTextViewTableViewCell (IntPtr handle) : base(handle)
		{
			
		}


		public override void AwakeFromNib ()
		{
			base.AwakeFromNib();

			PlaceholderLabel.Hidden = TextView.HasText;
		}


		public override void SetData (CellData cellData)
		{
			var detail = cellData.Get(DataKeys.Detail);
			var auxiliary = cellData.Get(DataKeys.Auxiliary);

			TextView.Text = detail;
			PlaceholderLabel.Hidden = TextView.HasText;
			PlaceholderLabel.Text = string.IsNullOrEmpty(detail) ? auxiliary : null;
		}


		[Export("textViewDidChange:")]
		public void Changed (UITextView textView)
		{
			PlaceholderLabel.Hidden = textView.HasText;
		}


		public void TextViewShouldBecomeFirstResponder ()
		{
			TextView.BecomeFirstResponder();
		}
	}
}

File: iOS/Code/Swift/bugTrap/bugTrapKit/Controllers/TableViewControllers/TableViewCells/BtTextViewTableViewCell.swift
//
//  BtTextViewTableViewCell.swift
//  bugTrap
//
//  Created by Colby L Williams on 6/30/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import UIKit

class BtTextViewTableViewCell: BtNewBugDetailsTableViewCell, UITextViewDelegate {
	
	@IBOutlet weak var TextView: UITextView!
	@IBOutlet weak var PlaceholderLabel: UILabel!
	
	override func awakeFromNib() {
		super.awakeFromNib()
		
		PlaceholderLabel.hidden = TextView.hasText()
	}
	
	func textViewDidChange(textView: UITextView) {
		PlaceholderLabel.hidden = TextView.hasText()
	}
	
	override func setData(title: String? = nil, detail: String? = nil, placeholder: String? = nil, auxiliary: String? = nil) {
		TextView.text = detail ?? ""
		PlaceholderLabel.hidden	= TextView.hasText()
		PlaceholderLabel.text = detail == nil || detail!.isEmpty ? auxiliary ?? "" : ""
	}
	
	func textViewShouldBecomeFirstResponder() {
		TextView.becomeFirstResponder()
	}
}